packages/pages/src/dev/server/ssr/findMatchingStaticTemplate.ts

```diff
@@ -1,26 +1,31 @@
 import { ViteDevServer } from "vite";
 import {
   getAllLocalData,
-  getLocalData,
+  readLocalDataFile,
   staticPageCriterion,
 } from "./getLocalData.js";
-import {
-  TemplateModuleInternal,
-  convertTemplateModuleToTemplateModuleInternal,
-} from "../../../common/src/template/internal/types.js";
-import { loadViteModule } from "./loadViteModule.js";
-import {
-  TemplateModule,
-  TemplateRenderProps,
-} from "../../../common/src/template/types.js";
+import { TemplateModuleInternal } from "../../../common/src/template/internal/types.js";
+import { TemplateRenderProps } from "../../../common/src/template/types.js";
 import { propsLoader } from "./propsLoader.js";
-import { findTemplateModuleInternal } from "./findTemplateModuleInternal.js";
+import { loadTemplateModuleInternal } from "./findTemplateModuleInternal.js";
 
 export type StaticTemplateAndProps = {
   staticTemplateModuleInternal: TemplateModuleInternal<any, any>;
   props: TemplateRenderProps;
 };
 
+type TemplateAndLocalDataPaths = {
+  templateFilePath?: string;
+  localDataFilename?: string;
+  isStatic: boolean;
+};
+
+// A cache of slug to template/localData filepaths to avoid loading everything on every lookup.
+const slugToModuleAndLocalDataPaths = new Map<
+  string,
+  TemplateAndLocalDataPaths
+>();
+
 /**
  * Loops through all static templates. For each, finds all localData static files
```

Comment:
is this comment still accurate

Reply:
Yes. Really all that changed here is using a cache to do lookups quicker.